### PR TITLE
I have completed all the week 4 assignments

### PR DESCRIPTION
**Pull request by sumithalder**
Here you can find all the solutions with problem descriptions. Also, I have added a readme file with solutions for better understanding.

### CODE REVIEW

File: Week-4/Exercise 4.1/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+const customPromiseState = {
+
+    PENDING: 'PENDING',
+    RESOLVED: 'RESOLVED',
+    REJECTED: 'REJECTED'
+
+}
+class customPromise {
```
Comment (pratik24) [inline Week-4/Exercise 4.1/index.js:8]: When naming class it is good practice to start with Capital letter

File: Week-4/Exercise 4.2/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+var Person = function () {};
+
+Person.prototype.initilize = function (name, age) {
```
Comment (pratik24) [inline Week-4/Exercise 4.2/index.js:3]: Typo him.initialize

File: Week-4/Exercise 4.2/index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+	this.name = name;
+	this.age = age;
+
+};
+
+var Teacher = function () {
+
+	Person.call(this);
```
Comment (pratik24) [inline Week-4/Exercise 4.2/index.js:12]: I think this call is not needed

**Review by pratik24 — APPROVED** [review]

File: Week-4/Exercise 4.1/index.js (hunk 1 above)
Comment (sumithalder) [reply Week-4/Exercise 4.1/index.js]: Done

File: Week-4/Exercise 4.2/index.js (hunk 3 above)
Comment (sumithalder) [reply Week-4/Exercise 4.2/index.js]: Done

**Pull request merged**